.github/workflows/functional-test.yaml

```diff
@@ -0,0 +1,128 @@
+    runs-on: ubuntu-latest
+    env:
+      PROJECT_ID: model-159019
+    steps:
+
+      # ---------------------
+      # Common steps that we repeat between workflows.
+      # I couldn't find an equivalent of CircleCI's templates; can we consolidate by turning this into an action?
+      - name: Code Checkout
+        uses: actions/checkout@v2
+
+      - name: Sub modules checkout
+        env:
+          SSH_KEY_FOR_SUBMODULE: ${{secrets.SUBMODULE_GITHUB_KEY}}
+        #the step below set the ssh key for the run
+        run: |
+          mkdir -p /home/runner/.ssh && touch /home/runner/.ssh/id_rsa && echo "$SSH_KEY_FOR_SUBMODULE" > $HOME/.ssh/id_rsa && chmod 600 $HOME/.ssh/id_rsa &&  git submodule sync && git submodule --quiet update --init && echo `git rev-parse --short=8 HEAD ` > TAG.txt
+
+      - name: Login to GCP
+        # Setup gcloud CLI
+        uses: google-github-actions/setup-gcloud@v0.2.0
+        with:
+          project_id: ${{ env.PROJECT_ID }}
+          service_account_key: ${{ secrets.GOOGLE_CREDENTIALS }}
+          version: "297.0.1" # Fix https://github.com/google-github-actions/setup-gcloud/issues/128
+      # Configure Docker to use the gcloud command-line tool as a credential
+      # helper for authentication
+      - run: |-
+          gcloud --quiet auth configure-docker
+
+      - name: Login to GCR
+        uses: docker/login-action@v1
+        with:
+          registry: gcr.io
+          username: _json_key
+          password: ${{ secrets.GOOGLE_CREDENTIALS }}
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@master
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-graphhopper-buildx-${{ github.sha}}
+          restore-keys: |
+            ${{ runner.os }}-graphhopper-buildx
+      # ---------------------
+
+      - name: build the server
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          push: true
+          file: Dockerfile.server
+          tags: "us.gcr.io/model-159019/gh:${{ github.sha }}-server-dev"
+          build-args: |
+            BASE_IMAGE=us.gcr.io/model-159019/gh:${{ github.sha }}-dev
+
+      - name: Install golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.16.6'
+
+      - name: Install grpcurl
+        run: |
+          go get github.com/fullstorydev/grpcurl/...
+          go install github.com/fullstorydev/grpcurl/cmd/grpcurl@latest
+          grpcurl -version
+
+      - name: run functional test
+        run:  bash functional_test.sh ${{ github.sha }}-server-dev
+  update-tags:
+    runs-on: ubuntu-latest
+    needs: functional-tests
+    if: github.ref == 'refs/heads/original-direction' && github.event.workflow_run.conclusion == 'success'
```

Comment:
Maybe add a small note/comment explicitly saying we don't want to run on `workflow_dispatch`?